dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -1022,6 +1048,39 @@
 		</ItemGroup>
 	</Target>
 
+	<Target Name="_CreateAOTDedupAssembly"
+			Condition="'$(_RunAotCompiler)' == 'true'">
+		<PropertyGroup>
+			<_DedupAssembly>$(IntermediateLinkDir)aot-instances.dll</_DedupAssembly>
+		</PropertyGroup>
+		<WriteLinesToFile File="$(_IntermediateNativeLibraryDir)/aot-instances.cs" Overwrite="true" Lines="" WriteOnlyWhenDifferent="true" />
+		<Csc Sources="$(_IntermediateNativeLibraryDir)\aot-instances.cs"
+		OutputAssembly="$(_DedupAssembly)"
+		TargetType="library"
+		Deterministic="true"
+		References="@(ReferencePath)"
+		ToolExe="$(CscToolExe)"
+		ToolPath="$(CscToolPath)" />
+		<Delete Files="$(_IntermediateNativeLibraryDir)aot-instances.cs" />
+
+		<ItemGroup>
+			<_Assemblies Include="$(_DedupAssembly)">
+				<AOTAssembly>_AOTAssembly</AOTAssembly>
+				<Arch>_Arch</Arch>
+				<Arguments>_Arguments</Arguments>
+				<ProcessArguments>_ProcessArguments</ProcessArguments>
+				<ObjectFile>_ObjectFile</ObjectFile>
+				<Abi>_Abi</Abi>
+				<LLVMFile>_LLVMFile</LLVMFile>
+				<AOTData>_AOTData</AOTData>
+			</_Assemblies>
+		</ItemGroup>
+
+		<CreateDedupItem Assemblies="@(_Assemblies)" DedupAssembly="$(_DedupAssembly)">
+			<Output ItemName="_AssembliesToAOT" TaskParameter="OutAssemblies" />
+		</CreateDedupItem>
```

Comment:
I think Ivan's idea here is good, it would simplify this a bit.

Reply:
When the dedup assembly is added to `ManagedAssemblyToLink` and `TrimmerRootAssembly`, MySingleView application crashes with `_ZNK3icu6number23NumberFormatterSettingsINS0_24LocalizedNumberFormatterEE10toSkeletonER10UErrorCode`. According to the https://github.com/xamarin/xamarin-macios/issues/17708, it could happen due to a missing assembly in the bundle. Something is missing during the build, as aot-instances.dll is not included in the bundle.

```terminal
ls -la bin/Release/net8.0-ios/ios-arm64/MySingleView.app
total 29744
drwxr-xr-x   21 miloskotlar  staff      672 Mar 14 11:33 .
drwxr-xr-x  211 miloskotlar  staff     6752 Mar 14 11:33 ..
-rw-r--r--    1 miloskotlar  staff     1066 Mar 14 11:32 Info.plist
-rw-r--r--    1 miloskotlar  staff    17432 Mar 14 11:32 Microsoft.iOS.aotdata.arm64
-rw-r--r--    1 miloskotlar  staff    44032 Mar 14 11:32 Microsoft.iOS.dll
-rwxr-xr-x    1 miloskotlar  staff  7486432 Mar 14 11:33 MySingleView
-rw-r--r--    1 miloskotlar  staff     1048 Mar 14 11:32 MySingleView.aotdata.arm64
-rw-r--r--    1 miloskotlar  staff     5120 Mar 14 11:32 MySingleView.dll
-rw-r--r--    1 miloskotlar  staff        8 Mar 14 11:32 PkgInfo
-rw-r--r--    1 miloskotlar  staff   408616 Mar 14 11:32 System.Private.CoreLib.aotdata.arm64
-rw-r--r--    1 miloskotlar  staff   436224 Mar 14 11:32 System.Private.CoreLib.dll
-rw-r--r--    1 miloskotlar  staff      792 Mar 14 11:32 System.Runtime.InteropServices.aotdata.arm64
-rw-r--r--    1 miloskotlar  staff     7680 Mar 14 11:32 System.Runtime.InteropServices.dll
-rw-r--r--    1 miloskotlar  staff      776 Mar 14 11:32 System.Runtime.aotdata.arm64
-rw-r--r--    1 miloskotlar  staff     5120 Mar 14 11:32 System.Runtime.dll
drwxr-xr-x    3 miloskotlar  staff       96 Mar 14 11:33 _CodeSignature
-rw-r--r--    1 miloskotlar  staff   350416 Mar 14 11:32 aot-instances.aotdata.arm64
-rw-r--r--    1 miloskotlar  staff      181 Mar 14 11:32 archived-expanded-entitlements.xcent
-rw-r--r--    1 miloskotlar  staff  4558655 Mar 14 11:33 embedded.mobileprovision
-rwxr--r--    1 miloskotlar  staff  1859648 Feb 16 21:41 icudt.dat
-rw-r--r--    1 miloskotlar  staff      983 Mar 14 11:32 runtimeconfig.bin
```

Reply:
It might be that you will have to include the assembly in `ResolvedFileToPublish` ItemGroup (not sure if explicitly, or there is a Target/Task which will do it for you).